src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -184,150 +184,199 @@ public static object ToBlittableSupportedType(object value, bool flattenArrays =
             return ToBlittableSupportedType(value, value, flattenArrays, forIndexing, 0, engine, context);
         }
 
-        private static object ToBlittableSupportedType(object root, object value, bool flattenArrays, bool forIndexing, int recursiveLevel, Engine engine, JsonOperationContext context)
-        {
-            if (recursiveLevel > MaxAllowedRecursiveLevelForType)
-                NestingLevelTooDeep(root);
 
-            if (value is JsValue js)
-            {
-                if (js.IsNull())
-                {
-                    if (forIndexing && js is DynamicJsNull dynamicJsNull)
-                        return dynamicJsNull.IsExplicitNull ? DynamicNullObject.ExplicitNull : DynamicNullObject.Null;
+        private enum BlittableSupportedReturnType : int
```

Comment:
Why the duplication with TypeConverter on the client?

Reply:
Because it was like that. The code is actually slightly different, so I didn't dare to unify them.